source/agora/consensus/protocol/Nominator.d

```diff
@@ -306,7 +339,10 @@ extern(D):
     public void stopNominationRound (Height height) @safe nothrow
     {
         this.is_nominating = false;
-        () @trusted { this.scp.stopNomination(height); }();
+        () @trusted {
+            if (this.scp != null)
+                this.scp.stopNomination(height);
```

Comment:
Can this ever be called if the node doesn't have an SCP object ?

Reply:
It happens. I think it's because an `outsider validator` exits without having created its SCP.

Reply:
the related error log.
```
Running build/agora-unittests
0   agora-unittests                     0x00000001056f0cbb _D4core7runtime9backtraceFNbPPviZi + 75
1   agora-unittests                     0x00000001056f0c4a _D4core7runtime18runModuleUnitTestsUZ19unittestSegvHandlerUNbiPSQCk3sys5posix6signal9siginfo_tPvZv + 58
2   libsystem_platform.dylib            0x00007fff6d40c5fd _sigtramp + 29
3   agora-unittests                     0x000000010505a316 _ZNKSt3__16__treeINS_12__value_typeIyNS_10shared_ptrIN7stellar4SlotEEEEENS_19__map_value_compareIyS6_NS_4lessIyEELb1EEENS_9allocatorIS6_EEE6__rootEv + 22
4   agora-unittests                     0x000000010505d278 _ZNSt3__16__treeINS_12__value_typeIyNS_10shared_ptrIN7stellar4SlotEEEEENS_19__map_value_compareIyS6_NS_4lessIyEELb1EEENS_9allocatorIS6_EEE4findIyEENS_15__tree_iteratorIS6_PNS_11__tree_nodeIS6_PvEElEERKT_ + 40
5   agora-unittests                     0x0000000105055d2d _ZNSt3__13mapIyNS_10shared_ptrIN7stellar4SlotEEENS_4lessIyEENS_9allocatorINS_4pairIKyS4_EEEEE4findERS9_ + 29
6   agora-unittests                     0x000000010505563d _ZN7stellar3SCP7getSlotEyb + 77
7   agora-unittests                     0x0000000105055943 _ZN7stellar3SCP14stopNominationEy + 35
8   agora-unittests                     0x00000001049ead87 _D5agora9consensus8protocol9NominatorQk19stopNominationRoundMFNbNfSQCn6common5Types6HeightZ9__lambda2MFNbNeZv + 23
9   agora-unittests                     0x00000001046f58c8 _D5agora9consensus8protocol9NominatorQk19stopNominationRoundMFNbNfSQCn6common5Types6HeightZv + 56
10  agora-unittests                     0x00000001046eec8b _D5agora4node9ValidatorQk15onAcceptedBlockMFNfIKSQBv9consensus4data5BlockQgbZv + 91
11  agora-unittests                     0x0000000104751347 _D5agora4node6LedgerQh11acceptBlockMFNfIKSQBo9consensus4data5BlockQgAyamZb + 551
12  agora-unittests                     0x00000001048f3b42 _D5agora4node8FullNodeQj9addBlocksMFNfAxSQBn9consensus4data5BlockQgAxSQCqQBdQw12PreImageInfoQoZSQDq6common5Types6Height + 226
13  agora-unittests                     0x00000001047c0a10 _D5agora7network7Manager14NetworkManager13getBlocksFromMFNbSQCg6common5Types6HeightMDFNfAxSQDl9consensus4data5BlockQgAxSQEoQBdQw12PreImageInfoQoZQDiZv + 912
14  agora-unittests                     0x00000001046ed2f3 _D5agora4node8FullNodeQj11catchupTaskMFNbZv + 147
15  agora-unittests                     0x0000000105200397 _D6geod249LocalRest5Timer3runMFZv + 55
16  agora-unittests                     0x00000001056f404d _D4core6thread7context8Callable6opCallMFZv + 61
17  agora-unittests                     0x00000001056f41d9 _D4core6thread5fiber5Fiber3runMFZv + 25
18  agora-unittests                     0x00000001056f4116 fiber_entryPoint + 150
```